CDS.Tests/Collections/TernarySearchTrieTests.cs

```diff
@@ -0,0 +1,108 @@
+﻿using CDS.Collections;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+
+namespace CDS.Tests.Collections
+{
+    public class TernarySearchTrieTests
+    {
+        [Fact]
+        public void Test_Put_Given_One_Element_Trie_Count_Equals_One()
+        {
+            var trie = new TernarySearchTrie<String>();
+
+            trie.Put("test", "val");
+
+            Assert.Equal(1, trie.Count);
+        }
+
+        [Fact]
+        public void Test_Put_Given_Five_Elements_Trie_Count_Equals_Five()
```

Comment:
This test method and the above can be combined to a single method with `[Theory]` and `[InlineData]`